controller/src/main/java/io/pravega/controller/server/bucket/ZooKeeperBucketManager.java

```diff
@@ -73,9 +88,15 @@ public void startBucketOwnershipListener() {
                     break;
                 case CHILD_REMOVED:
                     int bucketId = Integer.parseInt(ZKPaths.getNodeFromPath(event.getData().getPath()));
-                    RetryHelper.withIndefiniteRetriesAsync(() -> tryTakeOwnership(bucketId),
-                            e -> log.warn("{}: exception while attempting to take ownership for bucket {}: {}", getServiceType(),
-                                    bucketId, e.getMessage()), getExecutor());
+                    bucketStore.getBucketsForController(processId, getServiceType())
+                            .thenAccept(buckets -> {
+                                log.debug("{} : Buckets assigned to controller {} are {}", getServiceType(), processId, buckets);
+                                if (buckets.contains(bucketId)) {
```

Comment:
isnt this blocked covered by tests?

Reply:
This listener is redundant in new the approach. Removed this listener as in new approach, controller is dependent on buckets to controller mapping.